components/script/dom/underlyingsourcecontainer.rs

```diff
@@ -148,24 +140,16 @@ impl UnderlyingSourceContainer {
         controller: Controller,
         can_gc: CanGc,
     ) -> Option<Rc<Promise>> {
-        if let UnderlyingSourceType::Js(source) = &self.underlying_source_type {
+        if let UnderlyingSourceType::Js(source, this_obj) = &self.underlying_source_type {
             if let Some(start) = &source.start {
                 let cx = GlobalScope::get_cx();
-                rooted!(in(*cx) let mut this_object = ptr::null_mut::<JSObject>());
-                // TODO: move this into `bindings`.
-                unsafe {
-                    let obj = JS_NewObject(*cx, ptr::null_mut());
-                    assert!(!obj.is_null());
-                    this_object.set(obj);
-                    source.to_jsobject(*cx, this_object.handle_mut());
-                }
-                let this_handle = this_object.handle();
+                rooted!(in(*cx) let mut this_obj = this_obj.get());
```

Comment:
Why is this root necessary?

Reply:
It is because `Call_` expects a `js::rust::Handle`, and from the heap you get ` js::jsapi::Handle`. There doesn't seem to be an easy to create a stand-alone `js::rust::Handle` because [`new`](https://github.com/servo/mozjs/blob/f68e385deb685e8e74f50e71d878d0889903e378/mozjs/src/gc/root.rs#L107) is private.

Reply:
You want https://doc.servo.org/mozjs/gc/root/struct.Handle.html#method.from_raw

Reply:
Thank you.